src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -512,9 +514,14 @@ private void Write(TaskParameterEventArgs e)
             WriteTaskItemList(e.Items, e.LogItemMetadata);
         }
 
-        private void WriteBuildEventArgsFields(BuildEventArgs e, bool writeMessage = true)
+        private void WriteBuildEventArgsFields(BuildEventArgs e, bool writeMessage = true, bool writeLineAndColumn = false)
         {
             var flags = GetBuildEventArgsFieldFlags(e, writeMessage);
+            if (writeLineAndColumn)
+            {
+                flags |= BuildEventArgsFieldFlags.LineNumber | BuildEventArgsFieldFlags.ColumnNumber;
+            }
```

Comment:
nit: Why not move this into `GetBuildEventArgsFieldFlags`?

Reply:
I can’t exactly remember but I slightly preferred this version, perhaps to avoid dragging that extra bool parameter around.